src/IssueViz.UnitTests/Selection/AnalysisIssueSelectionServiceTests.cs

```diff
@@ -274,41 +322,83 @@ public void SelectedIssueChanged_ChangesSelectedFlow()
             testSubject.SelectedFlow.Should().BeNull();
 
             // Set issue to different value
-            var secondIssue = GetIssueWithFlow(out var secondIssueFirstFlow);
+            var secondIssueFirstFlow = CreateFlow();
+            var secondIssue = CreateIssue(secondIssueFirstFlow);
             testSubject.SelectedIssue = secondIssue;
             eventHandler.Verify(x => x(testSubject, It.Is((SelectionChangedEventArgs args) => args.SelectedFlow == secondIssueFirstFlow)), Times.Once());
             testSubject.SelectedFlow.Should().Be(secondIssueFirstFlow);
 
             eventHandler.VerifyNoOtherCalls();
         }
 
-        private static IAnalysisIssueFlowVisualization GetFlowWithLocation(out IAnalysisIssueLocationVisualization firstLocation)
+        [TestMethod]
+        public void IssueSelectionServiceEvent_IssueHasSecondaryLocations_SelectedIssueIsSet()
         {
-            firstLocation = Mock.Of<IAnalysisIssueLocationVisualization>();
-            var otherLocation = Mock.Of<IAnalysisIssueLocationVisualization>();
+            var location = Mock.Of<IAnalysisIssueLocationVisualization>();
+            var flow = CreateFlow(location);
+            var issue = CreateIssue(flow);
 
-            var mockFlow = new Mock<IAnalysisIssueFlowVisualization>();
-            mockFlow
-                .Setup(x => x.Locations)
-                .Returns(new[] {firstLocation, otherLocation});
+            var selectionService = new Mock<IIssueSelectionService>();
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
 
-            return mockFlow.Object;
+            RaiseSelectedIssueChangedEvent(selectionService, issue);
+
+            testSubject.SelectedIssue.Should().Be(issue);
+            testSubject.SelectedFlow.Should().Be(flow);
+            testSubject.SelectedLocation.Should().Be(location);
         }
 
-        private IAnalysisIssueVisualization GetIssueWithFlow(out IAnalysisIssueFlowVisualization firstFlow)
+        [TestMethod]
+        public void IssueSelectionServiceEvent_IssueHasNoSecondaryLocations_SelectedIssueIsCleared()
         {
-            firstFlow = Mock.Of<IAnalysisIssueFlowVisualization>();
-            var otherFlow = Mock.Of<IAnalysisIssueFlowVisualization>();
+            var issue = CreateIssue();
 
-            var mockIssue = new Mock<IAnalysisIssueVisualization>();
-            mockIssue
-                .Setup(x => x.Flows)
-                .Returns(new[] { firstFlow, otherFlow });
+            var selectionService = new Mock<IIssueSelectionService>();
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
 
-            return mockIssue.Object;
+            RaiseSelectedIssueChangedEvent(selectionService, issue);
+
+            testSubject.SelectedIssue.Should().BeNull();
```

Comment:
Was `testSubject.SelectedIssue` ever set to anything other than `null`?
If not, this test doesn't test that the issue is cleared. Same for the next test.